mountpoint-s3-fs/src/prefetch/part_stream.rs

```diff
@@ -437,11 +443,14 @@ fn read_from_request<'a, Client: ObjectClient + 'a>(
             if excess_bytes > 0 {
                 metrics::histogram!("s3.client.read_window_excess_bytes").record(excess_bytes as f64);
             }
-            // Blocks if read window increment if it's not enough to read the next offset
+
+            // When we detect an updated read window end offset, pass this signal on to the S3 client.
+            // TODO:
+            //   It does not make sense to 'block' here. In reality, we don't actually block here anyway.
```

Comment:
alternatively we may have a separate async task responsible for advancing the read window based on values from the channel, while this task will yield data as soon as it is available in request

the issue with `backpressure controller should communicate directly with the client` is that the requests do not yet exist when `BackpressureController` is instantiated, so we will have to spawn both requests early (with zero read window?) in order to have `client_backpressure_handle` in the right place

Reply:
Makes sense, I was expecting there to be some juggling to do. I do wonder if it would be better yet to avoid introducing another async task, but I think we need to think about it a bit more.

I'll rework the comment a bit more so it's less prescriptive and more exploring ideas.